server/routes/api/stats.py

```diff
@@ -414,65 +414,212 @@ def get_place_stat_date_within_place():
 
     Args:
-        stat_set: set of statistical variable data points with the format: 
+        series_response: the response from a dc.series_within call.
+    
+    Returns:
+        a dict of sv to dict of facet id to facet information:
             {
-                data: {
-                    [svDcid]: {
-                        stat: {
-                            [placeDcid]: {
-                                date: "",
-                                value: "",
-                                metaHash: ""
-                            } ,
-                            ...
-                        }
+                [sv]: {
+                    [facet1]: {
+                        "importName": "Census",
+                        "observationPeriod": "P1Y",
+                        ...
                     },
                     ...
                 },
-                metadata: {
-                    [metaHash]: {
-                        provenanceUrl: "",
+                ...
+            }
+    """
+    facets_by_variable = {}
+    facets = series_response.get("facets", {})
+    for sv_obs in series_response.get("observationsByVariable", []):
+        sv = sv_obs.get("variable", "")
+        facets_by_variable[sv] = {}
+        for place_obs in sv_obs.get("observationsByEntity", []):
+            for facet_obs in place_obs.get("seriesByFacet", []):
+                facet = str(facet_obs.get("facet", ""))
+                if facet and not facet in facets_by_variable[sv]:
+                    facets_by_variable[sv][facet] = facets.get(facet, {})
+    return facets_by_variable
+
+
+def get_facets_by_variable_points(points_response):
+    """
+    Gets the available facets for each sv in an api response for points_within.
+
+    Args:
+        points_response: the response from a dc.points_within call.
+    
+    Returns:
+        a dict of sv to dict of facet id to facet information:
+            {
+                [sv]: {
+                    [facet1]: {
+                        "importName": "Census",
+                        "observationPeriod": "P1Y",
                         ...
                     },
                     ...
-                }
+                },
+                ...
             }
+    """
+    facets_by_variable = {}
+    facets = points_response.get("facets", {})
+    for sv_obs in points_response.get("observationsByVariable", []):
+        sv = sv_obs.get("variable", "")
+        facets_by_variable[sv] = {}
+        for place_obs in sv_obs.get("observationsByEntity", []):
+            for facet_obs in place_obs.get("pointsByFacet", []):
+                facet = str(facet_obs.get("facet", ""))
+                if not facet in facets_by_variable[sv]:
+                    facets_by_variable[sv][facet] = facets.get(facet, {})
+    return facets_by_variable
+
+
+@bp.route('/api/stats/facets/within-place')
+@cache.cached(timeout=3600 * 24, query_string=True)
```

Comment:
Since the dc.xxx API call are all cached, here does not really need to use @cached for caching, since there are no computation inside this function.

With that, can change this route to POST, which make will the client code more readable by using req_json instead of building long url string

Reply:
done